src/sage/schemes/chow/ring_element.py

```diff
@@ -0,0 +1,503 @@
+        """
+        self._parentchowring = parent
+        # Reduce is broken (and not needed) for parents with no generators.
+        red = False if parent.ngens() == 0 else True
+        QuotientRingElement.__init__(self, parent, rep, reduce=red)
+
+    def _richcmp_(self, other, op):
+        r"""
+        Compares two elements.
+
+        EXAMPLES::
+
+            sage: A.<H,E> = ChowRing(['H', 'E'], [1, 1], ['E*H', 'E^2+H^2'])
+            sage: H^2 == -E^2
+            True
+            sage: 2 + H^2 == E - E^2
+            False
+
+        TESTS::
+
+            sage: A = ChowRing()
+            sage: A(5) == A(0)
+            False
+        """
+        A = self._parentchowring
+        if A.ngens() == 0:
+            return richcmp(QQ(str(self)), QQ(str(other)), op)
+        return QuotientRingElement._richcmp_(self, other, op)
+
+    @cached_method
+    def by_degrees(self):
+        r"""
+        Returns this Chow ring element as a list `[e_0,\dots,e_{d}]` where `e_k`
+        has degree `k`.
+
+        EXAMPLES::
+
+            sage: A.<H,E> = ChowRing(['H', 'E'], [1, 1], ['E*H', 'E^2+H^2'])
+            sage: (-2 - E + 3*H - 4*H^2).by_degrees()
+            [-2, 3*H - E, 4*E^2]
+            sage: (-E + 3*H - 4*H^2).by_degrees()
+            [0, 3*H - E, 4*E^2]
+            sage: (-2 - E + 3*H).by_degrees()
+            [-2, 3*H - E]
+
+        TESTS::
+
+            sage: A = ChowRing()
+            sage: A(5).by_degrees()
+            [5]
+        """
+        A = self._parentchowring
+        # The computation is done in the cover ring in order to be able
+        # to use the methods monomials and coefficients, then the list
+        # is returned as a list of elements of the parent chow ring.
+        monoms, coeffs = self.lift().monomials(), self.lift().coefficients()
+        l = len(monoms)
+        d = max([monoms[i].degree() for i in range(l)]) if l else 0
+        p = [0] * (d + 1)
+        for i in range(len(monoms)):
+            k = monoms[i].degree()
+            if k <= d:
+                p[k] += coeffs[i] * monoms[i]
+        return [A(v) for v in p]
+
+    def truncate(self, a=0, b=0):
+        r"""
+        Return this Chow ring element truncated in degrees below a and above b
+        (strictly).
+
+        INPUT:
+
+        - ``a`` -- an optional integer (defaults to 0)
+
+        - ``b`` -- an optional integer (defaults to 0)
+
+        OUTPUT:
+
+        - a Chow ring element.
+
+        EXAMPLES::
+
+            sage: A.<H,E> = ChowRing(['H', 'E'], [1, 1], ['E*H', 'E^2+H^2'])
+            sage: x = -2 - E + 3*H - 4*H^2; x.by_degrees()
+            [-2, 3*H - E, 4*E^2]
+            sage: x.truncate()
+            -2
+            sage: x.truncate(0, 1)
+            3*H - E - 2
+            sage: x.truncate(0, 2)
+            4*E^2 + 3*H - E - 2
+            sage: x.truncate(1, 2)
+            4*E^2 + 3*H - E
+        """
+        return sum([x for x in self.by_degrees()[a:b + 1]])
```

Comment:
```suggestion
        return self.parent().sum(x for x in self.by_degrees()[a:b + 1])
```
In case the result is 0 in order to have the correct parent.